web/src/favicon.ts

```diff
@@ -6,9 +6,15 @@ import render_favicon_svg from "../templates/favicon.svg.hbs";
 import * as blueslip from "./blueslip";
 import favicon_font_url from "./favicon_font_url!=!url-loader!font-subset-loader2?glyphs=0123456789KMGT∞!source-sans/TTF/SourceSans3-Bold.ttf";
 
-let favicon_state;
+let favicon_state: {image: HTMLImageElement; url: string} | undefined;
 
-function load_and_set_favicon(rendered_favicon) {
+function set_favicon(): void {
+    if (favicon_state !== undefined) {
+        $("#favicon").attr("href", favicon_state.url);
+    }
+}
```

Comment:
It would be better to do the reordering in a separate commit. I think there is no need to reorder the function now that [we might be](https://chat.zulip.org/#narrow/stream/6-frontend/topic/no-use-before-define/near/1530797) loosening the eslint rule that checks for the order of definition and usage.